src/integrationTest/groovy/wooga/gradle/unity/UnityPluginIntegrationSpec.groovy

```diff
@@ -336,4 +343,30 @@ class UnityPluginIntegrationSpec extends UnityIntegrationSpec {
         then:
         result.wasExecuted("generateSolution")
     }
+
+    @Unroll
+    def "runs addUPMPackages task if there are packages to add"() {
+        given:
+        buildFile << """
+            unity {
+                enableTestCodeCoverage = ${testCoverageEnabled}
+                upmPackages = ${wrapValueBasedOnType(packagesToInstall, Map)}
+            }
+        """
+        when:
+        def result = runTasks("test")
+
+        then:
+        shouldRun ?
+                result.wasExecuted("addUPMPackages") :
+                result.standardOutput.contains("Task :addUPMPackages SKIPPED")
```

Comment:
there is a `wasSkipped` check to test this.

Reply:
For some reason it just doesn't work.

Reply:
strange. because wasExecuted returns true also for skipped tasks.